lib/services/saml.go

```diff
@@ -320,7 +320,7 @@ func (o *SAMLConnectorV2) Equals(other SAMLConnector) bool {
 		if (a.RoleTemplate != nil && b.RoleTemplate == nil) || (a.RoleTemplate == nil && b.RoleTemplate != nil) {
 			return false
 		}
-		if a.RoleTemplate != nil && !a.RoleTemplate.Equals(b.RoleTemplate) {
+		if a.RoleTemplate != nil && !a.RoleTemplate.Equals(b.RoleTemplate.V3()) {
```

Comment:
Yeah, I didn't focus too much on this logic because it's going to go away anyway. I also removed some tests for role templating because it's also going to be removed soon anyway.